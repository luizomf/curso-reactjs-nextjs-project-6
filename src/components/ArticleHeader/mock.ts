import { ArticleHeaderProps } from '.';
import { data } from '../../api/dados.json';

const { title, excerpt, cover, author, categories, createdAt } = data.posts[0];

export default {
  title,
  excerpt,
  cover,
  author,
  categories,
  createdAt,
} as ArticleHeaderProps;
