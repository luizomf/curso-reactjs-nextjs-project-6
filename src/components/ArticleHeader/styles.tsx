import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: ${theme.font.sizes.medium};
  `}
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
