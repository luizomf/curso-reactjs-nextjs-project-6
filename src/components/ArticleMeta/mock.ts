import { ArticleMetaProps } from '.';

export default {
  createdAt: '2021-03-02T15:16:09.246+00:00',
  author: {
    id: '123456789',
    slug: 'otavio-miranda',
    displayName: 'Ot√°vio Miranda',
  },
  categories: [
    {
      id: 'asdslgyh65',
      displayName: 'Tech',
      slug: 'tech',
    },
    {
      id: 'asdslgyh66',
      displayName: 'JS',
      slug: 'javascript',
    },
  ],
} as ArticleMetaProps;
