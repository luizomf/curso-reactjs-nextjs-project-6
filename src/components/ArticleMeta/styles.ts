import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkerGray};
    font-style: italic;

    .categories span::after {
      content: ', ';
    }

    .categories span:last-child::after {
      content: '';
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }
  `}
`;
