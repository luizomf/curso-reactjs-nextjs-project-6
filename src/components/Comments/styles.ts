import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    max-width: ${theme.sizes.content};
    width: 100%;
    margin: ${theme.spacings.large} auto;
    padding: 0 ${theme.spacings.large};
  `}
`;
