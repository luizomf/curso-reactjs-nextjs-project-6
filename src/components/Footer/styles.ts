import styled, { css } from 'styled-components';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.1rem solid ${theme.colors.mediumGray};

    a {
      color: inherit;
      text-decoration: none;
    }
  `}
`;
