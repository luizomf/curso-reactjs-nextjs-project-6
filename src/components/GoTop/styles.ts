import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background: ${theme.colors.primary};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    z-index: 6;
  `}
`;
