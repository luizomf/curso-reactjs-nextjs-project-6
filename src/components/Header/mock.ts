import { HeaderProps } from '.';

export default {
  blogName: 'Ot√°vio Miranda',
  blogDescription:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud.',
  logo: '/assets/images/otavio-miranda.png',
  showText: true,
} as HeaderProps;
