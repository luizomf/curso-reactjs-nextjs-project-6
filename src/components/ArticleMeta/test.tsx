import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { ArticleMeta, ArticleMetaProps } from '.';

import mock from './mock';

const props: ArticleMetaProps = mock;

describe('<ArticleMeta />', () => {
  it('should render author and category links', () => {
    renderTheme(<ArticleMeta {...props} />);

    expect(
      screen.getByRole('link', { name: 'OtÃ¡vio Miranda' }),
    ).toHaveAttribute('href', '/author/otavio-miranda');
    expect(screen.getByRole('link', { name: 'Tech' })).toHaveAttribute(
      'href',
      '/category/tech',
    );
    expect(screen.getByRole('link', { name: 'JS' })).toHaveAttribute(
      'href',
      '/category/javascript',
    );
  });

  it('should format date', () => {
    renderTheme(<ArticleMeta {...props} />);

    expect(screen.getByText('2 de mar. de 2021')).toHaveAttribute(
      'datetime',
      props.createdAt,
    );
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<ArticleMeta {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with no author and categories', () => {
    const { container } = renderTheme(
      <ArticleMeta {...props} author={undefined} categories={undefined} />,
    );
    expect(container).toMatchSnapshot();
  });
});
