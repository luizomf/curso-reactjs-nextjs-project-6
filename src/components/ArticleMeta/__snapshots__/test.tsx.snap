// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleMeta /> should match snapshot 1`] = `
.c0 {
  font-size: 1.6rem;
  color: #AAAAAA;
  font-style: italic;
}

.c0 .categories span::after {
  content: ', ';
}

.c0 .categories span:last-child::after {
  content: '';
}

.c0 a {
  color: #dc143c;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

<div>
  <div
    class="c0"
  >
    <p>
      <span>
        Por 
      </span>
      <a
        href="/author/otavio-miranda"
      >
        Ot√°vio Miranda
      </a>
      <span
        class="separator"
      >
         | 
      </span>
      <time
        datetime="2021-03-02T15:16:09.246+00:00"
      >
        2 de mar. de 2021
      </time>
      <span
        class="separator"
      >
         | 
      </span>
      <span
        class="categories"
      >
        <span>
          <a
            href="/category/tech"
          >
            Tech
          </a>
        </span>
        <span>
          <a
            href="/category/javascript"
          >
            JS
          </a>
        </span>
      </span>
    </p>
  </div>
</div>
`;

exports[`<ArticleMeta /> should match snapshot with no author and categories 1`] = `
.c0 {
  font-size: 1.6rem;
  color: #AAAAAA;
  font-style: italic;
}

.c0 .categories span::after {
  content: ', ';
}

.c0 .categories span:last-child::after {
  content: '';
}

.c0 a {
  color: #dc143c;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

<div>
  <div
    class="c0"
  >
    <p>
      <time
        datetime="2021-03-02T15:16:09.246+00:00"
      >
        2 de mar. de 2021
      </time>
    </p>
  </div>
</div>
`;
